File: frontend/src/app/models/academic.models.ts
export interface Student {
  id: string;
  fullName: string;
  faculty: string;
  major: string;
  classCode: string;
  intakeYear: number;
  email: string;
}

export interface ScheduleItem {
  id: string;
  studentId: string;
  courseCode: string;
  courseName: string;
  lecturer: string;
  dayOfWeek: string;
  startPeriod: number;
  endPeriod: number;
  room: string;
  startDate: string;
  endDate: string;
}

export interface GradeRecord {
  id: string;
  studentId: string;
  courseCode: string;
  courseName: string;
  credits: number;
  processScore: number;
  finalScore: number;
  totalScore: number;
  letterGrade: string;
  semester: string;
  isRetakeNeeded: boolean;
}

export interface AcademicSummary {
  studentId: string;
  gpa10: number;
  gpa4: number;
  completedCredits: number;
  debtCredits: number;
  warningFlags: string[];
  retakeCourses: GradeRecord[];
}

export interface ChatMessage {
  id: string;
  sessionId: string;
  role: 'user' | 'assistant';
  content: string;
  createdAt: string;
}

export interface AuthUser {
  id: string;
  fullName: string;
  email: string;
  studentId: string;
}

export interface AuthResponse {
  accessToken: string;
  user: AuthUser;
  student: Student;
}

export interface RegisterRequest {
  fullName: string;
  email: string;
  password: string;
  studentId?: string;
}

export interface LoginRequest {
  email: string;
  password: string;
  rememberMe: boolean;
}

export interface ChatSession {
  id: string;
  studentId: string;
  title: string;
  createdAt: string;
  updatedAt: string;
}

export interface ChatCitation {
  source: string;
  title: string;
  excerpt: string;
}

export interface ChatCard {
  type: 'schedule' | 'grades' | 'student' | 'tool_results' | 'retrieval_results' | string;
  title: string;
  data: unknown;
}

export interface ChatResponse {
  sessionId: string;
  userMessage: ChatMessage;
  assistantMessage: ChatMessage;
  citations: ChatCitation[];
  cards: ChatCard[];
  status: string;
  route: string;
  needsClarification: boolean;
  clarificationQuestions?: string[];
  error?: unknown;
  agent?: unknown;
}

export interface ChatRequest {
  studentId: string;
  sessionId?: string;
  message: string;
}

File: backend/AcademicSupport.Api/Models/AcademicModels.cs
namespace AcademicSupport.Api.Models;

public sealed record ApiError(string Code, string Message);

public sealed record UserAccount(
    Guid Id,
    string FullName,
    string Email,
    string StudentId,
    DateTimeOffset CreatedAt);

public sealed record AuthUser(
    Guid Id,
    string FullName,
    string Email,
    string StudentId);

public sealed record RegisterRequest(
    string FullName,
    string Email,
    string Password,
    string? StudentId);

public sealed record LoginRequest(
    string Email,
    string Password,
    bool RememberMe);

public sealed record AuthResponse(
    string AccessToken,
    AuthUser User,
    Student Student);

public sealed record Student(
    string Id,
    string FullName,
    string Faculty,
    string Major,
    string ClassCode,
    int IntakeYear,
    string Email);

public sealed record ScheduleItem(
    string Id,
    string StudentId,
    string CourseCode,
    string CourseName,
    string Lecturer,
    string DayOfWeek,
    int StartPeriod,
    int EndPeriod,
    string Room,
    DateOnly StartDate,
    DateOnly EndDate);

public sealed record GradeRecord(
    string Id,
    string StudentId,
    string CourseCode,
    string CourseName,
    int Credits,
    decimal ProcessScore,
    decimal FinalScore,
    decimal TotalScore,
    string LetterGrade,
    string Semester,
    bool IsRetakeNeeded);

public sealed record AcademicSummary(
    string StudentId,
    decimal Gpa10,
    decimal Gpa4,
    int CompletedCredits,
    int DebtCredits,
    IReadOnlyList<string> WarningFlags,
    IReadOnlyList<GradeRecord> RetakeCourses);

public sealed record ChatSession(
    Guid Id,
    string StudentId,
    string Title,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt);

public sealed record ChatMessage(
    Guid Id,
    Guid SessionId,
    string Role,
    string Content,
    DateTimeOffset CreatedAt);

public sealed record CreateSessionRequest(string StudentId, string? Title);

public sealed record ChatRequest(
    string StudentId,
    Guid? SessionId,
    string Message);

public sealed record ChatCitation(
    string Source,
    string Title,
    string Excerpt);

public sealed record ChatCard(
    string Type,
    string Title,
    object Data);

public sealed record ChatResponse(
    Guid SessionId,
    ChatMessage UserMessage,
    ChatMessage AssistantMessage,
    IReadOnlyList<ChatCitation> Citations,
    IReadOnlyList<ChatCard> Cards,
    string Status = "answered",
    string Route = "agent",
    bool NeedsClarification = false,
    IReadOnlyList<string>? ClarificationQuestions = null,
    object? Error = null,
    object? Agent = null);
